pysoa/test/plugins/pytest.py

```diff
@@ -0,0 +1,535 @@
+            local_variables={
+                k: v for k, v in six.iteritems(lowest_test_case_frame.locals)
+                if k in locals_to_copy
+            },
+            fixture_source=test_case['fixture_source'],
+            test_source=test_case['source'],
+            raw_entry=lowest_test_case_frame._rawentry,
+        )
+        exception_info.traceback.append(extra_entry)
+
+
+# noinspection SpellCheckingInspection
+class ServicePlanFixtureTestTracebackEntry(TracebackEntry):
+    """
+    A special traceback entry for displaying the relevant test fixture file contents instead of Python code when a
+    fixture test case fails.
+    """
+    class Faker(object):
+        pass
+
+    def __init__(
+        self,
+        name,
+        line_number,
+        path,
+        local_variables,
+        fixture_source,
+        test_source,
+        raw_entry,
+    ):
+        super(ServicePlanFixtureTestTracebackEntry, self).__init__(raw_entry)
+
+        self._name = name
+        self.lineno = line_number - 1
+        self._path = path
+        self._locals = local_variables
+        self._fixture_source = Source(fixture_source)
+        self._test_source = test_source
+
+        self._frame = self.Faker()
+        self._frame.statement = self.statement
+        self._frame.getargs = lambda *_, **__: list(six.iteritems(local_variables))
+        self._frame.f_locals = local_variables
+        self._frame.code = self.Faker()
+        self._frame.code.path = path
+        self._frame.code.raw = self.Faker()
+        self._frame.code.raw.co_filename = str(path)
+
+    @property
+    def frame(self):
+        return self._frame
+
+    @property
+    def relline(self):
+        return self.lineno - self.getfirstlinesource()
+
+    @property
+    def statement(self):
+        return self._fixture_source[self.lineno]
+
+    @property
+    def path(self):
+        return self._path
+
+    def getlocals(self):
+        return self._locals
+    locals = property(getlocals, None, None, str('locals of underlying frame'))
+
+    def getfirstlinesource(self):
+        return max(self.lineno - 3, 0)
+
+    def getsource(self, astcache=None):
+        start = self.getfirstlinesource()
+        end = start + len(self._test_source) + 5
+        return self._fixture_source[start:end]
+    source = property(getsource, None, None, str('source code of failing test'))
+
+    def ishidden(self):
+        return False
+
+    def name(self):
+        return self._name
+    name = property(name, None, None, str('name of underlaying code'))
+
+    def __str__(self):
+        return '  File {path} line {line_number} (approximate) in {test}\n  {source}\n'.format(
+            path=self.path,
+            line_number=self.lineno + 1,
+            test=self.name,
+            source=self._test_source,
+        )
+
+    def __repr__(self):
+        return '<TracebackEntry {}:{}>'.format(self.path, self.lineno + 1)
+
+
+def pytest_collection_modifyitems(config, items):
```

Comment:
This sounds like it should be possible to do with normal marking instead.  Maybe the collector or something else can use applymarker() to mark the tests and then more usual mechanisms can be used to select these tests instead of having to implement all this yourself.